### PR TITLE
UI redesign

### LINKED ISSUE #709: Z-index of error messages is too low
## Expected Behavior
error message should be visible

## Current Behavior
it is not visible
image

## Possible Solution
set z-index higher

## Debug info
KeePassXC - 2.4.3
KeePassXC-Browser - 1.5.3
Operating system: Linux x86_64
Browser: Chrome/Chromium 78.0.3904.108

### PR DESCRIPTION
**Pull request by varjolintu**
This overrides the previous PR (which is not up-to-date) and updates the user interface without any build scripts or additional dependencies.

- New settings page design
- Glyphicons are replaced with Fork Awesome
- Updated Bootstrap from version 3.3.7 to 4.4.1
- Updated jQuery from version 3.3.1 to 3.4.1

Fixes #709.

TODO:
- [x] Settings page menu collapse
- [x] Testing

Light theme preview:
light_theme

Dark theme preview:
dark_theme

### CODE REVIEW

**Comment by varjolintu** [comment]
This works quite well, so let's hit this with the next release.

**Comment by phoerious** [comment]
I think the sidebar dark theme has too little contrast. I am not sure if you used the highlight colour or the link colour, but I wouldn't use it for the main navigation.

**Comment by varjolintu** [comment]
@phoerious Thanks for the feedback.

Current link color (#6CAC4D) shows contrast ratio 4.9:1.
old-color

If I adjust the link color to the same as in your theme PR (#429F14), the ratio is 4:1.
new-color

The highlight color would be between these two.

**Comment by phoerious** [comment]
Still, I would prefer the normal foreground colour for the navigation. The grey text above certainly does not fulfil any WCAG contrast requirements.

**Comment by varjolintu** [comment]
@phoerious I updated the colors. Sidebar now use fixed colors for text and links. I checked the contrast and it should be enough, also for links.

**Comment by droidmonkey** [comment]
Some initial findings:

1. "Configure Keyboard Shortcuts" goes to the wrong page on Edge, should go to edge://extensions/shortcuts

2. The currently selected tab indicator disappears when interacting with the right pane:
image

3. All table rows have a weird hover effect that implies interaction, but there is no interaction. Just removed the hovering and highlighting of table rows.
image

4. Tiny text! (Also very hard to read red text on gray, make it KPXC green)
image

5. Buttons are ever so slightly taller than input fields. Make the input fields taller.
image

**Comment by varjolintu** [comment]
@droidmonkey Necessary fixes are made.

**Comment by droidmonkey** [comment]
Number 1 and 2 still seem to be an issue.

**Comment by varjolintu** [comment]
@droidmonkey Edge support is now added. I cannot reproduce issue 2. Nothing happens on hover.

**Comment by droidmonkey** [comment]
I agree it stays dark gray but would prefer the white outline to stay in place as well.

**Comment by varjolintu** [comment]
@droidmonkey Rest of the changes made.

**Comment by xf-** [comment]
Nice work!

- Alert background in in mainbar
image
- Bootstrap Select style missing in site preferences
- ` <label for="manualUrl" data-i18n="optionsSitePreferencesManualAddText"></label>` form nesting is wrong (results in missing bprder-radius on right side)
- Maybe use responsive tables on tables to allow scrolling (AddOns are possible on Android)

(- favicon possible? - easier to identify the tab)

I had some trouble uploading more screenshots to github after the first one.

**Comment by varjolintu** [comment]
@xf- I fixed most of the issues. I will definitely take a look at the fonts.

jQuery isn't nize, and options + popup pages are still using it. But I want to get rid of it when it's possible.

**Comment by xf-** [comment]
Nice changes :)

Padding ion left is missing compared (help text) & both form elements in one form-group
image

Also padding vs filling credentials alert
image

**Comment by varjolintu** [comment]
@xf- Those are also fixed now.

**Comment by xf-** [comment]
image
Still some indents. 

I will add a change to my PR
image

**Comment by droidmonkey** [comment]
@xf- I am not seeing any indents at all.

image

**Comment by varjolintu** [comment]
@droidmonkey Changed the popup font. Thanks for the review.

**Comment by varjolintu** [comment]
I'm also updating `package.json` and `package-lock.json` files, plus uploading `.eslintrc` for the project.

File: keepassxc-browser/options/options.css [hunk 1]
```diff
@@ -141,44 +170,50 @@ h2+hr {
...
-    display: none;
-}
+    .collapse:not(.show) {
+        display: none !important;
+    }
 
-.panel-title {
-    font-size: 14px !important;
-}
+    .d-flex {
```
Comment (xf-) [inline keepassxc-browser/options/options.css]: d-none does that form bootstrap

File: keepassxc-browser/options/options.html [hunk 2]
```diff
@@ -2,644 +2,767 @@
 <html>
```
Comment (xf-) [inline keepassxc-browser/options/options.html]: Doctype missing

File: keepassxc-browser/options/options.html [hunk 3]
```diff
@@ -2,644 +2,767 @@
...
+                  <span data-i18n="optionsKeyboardShortcutsHeader"></span>
+                </div>
+                <div class="card-body">
+                  <p>
+                    <span data-i18n="contextMenuFillUsernameAndPassword"></span>: <span id="fill_username_password-shortcut">error</span><br />
+                    <span data-i18n="contextMenuFillPassword"></span>: <span id="fill_password-shortcut">error</span><br />
+                    <span data-i18n="contextMenuFillTOTP"></span>: <span id="fill_totp-shortcut">error</span><br />
+                    <span data-i18n="contextMenuShowPasswordGenerator"></span>: <span id="show_password_generator-shortcut">error</span>
+                  </p>
+                  <p id="chrome-only">
```
Comment (xf-) [inline keepassxc-browser/options/options.html]: like `chrome-only` could be reused and `id` is only allowed once per page. Most times a `class` is used.

File: keepassxc-browser/options/options.html [hunk 4]
```diff
@@ -2,644 +2,767 @@
...
+                    <span data-i18n="optionsSitePreferencesTabHelpTextFourth"></span>
+                  </p>
+                  <hr />
+
+                  <form class="was-validated">
+                    <div class="form-group">
+                        <label for="manualUrl" data-i18n="optionsSitePreferencesManualAddText"></label>
+                      <div class="input-group" id="manualUrlGroup">
+                        <input class="form-control form-control-sm" type="url" id="manualUrl" aria-label="Manual URL" pattern="ftp://.*|http://.*|https://.*" minlength="10" required>
+                        <div class="input-group-append" id="button-addon4">
```
Comment (xf-) [inline keepassxc-browser/options/options.html]: duplicate id in options - id is only allowed once per page

File: keepassxc-browser/options/options.html [hunk 5]
```diff
@@ -2,644 +2,767 @@
...
+                      Holder(s) under this license and clearly marked as such. This may
+                      include source files, build scripts and documentation.
+
+                      "Reserved Font Name" refers to any names specified as such after the
+                      copyright statement(s).
+
+                      "Original Version" refers to the collection of Font Software components as
+                      distributed by the Copyright Holder(s).
+
+                      "Modified Versi
```
Comment (xf-) [inline keepassxc-browser/options/options.html]: misses a closing div

File: keepassxc-browser/options/options.html [hunk 6]
```diff
@@ -2,644 +2,767 @@
...
+                    <br />
+                    <br />
+                    <div>
+                      <span data-i18n="optionsVersionRunning" i18n-placeholder="<em class='yourVersion'></em>"></span>
+                    </div>
+                    <div>
+                      <p><span data-i18n="optionsLatestVersion" i18n-placeholder="<em class='latestVersion'></em>"></span></p>
+                      <button class="btn btn-sm btn-success checkUpdateKeePassXC"><i class="fa fa-refresh" aria-hidden="true"></i><span data-i18n="optionsButtonUpdate"></span></button>
+                    </div>
+                    <p>
```
Comment (xf-) [inline keepassxc-browser/options/options.html]: p is not allowed to contain block elements

File: keepassxc-browser/options/options.html [hunk 7]
```diff
@@ -2,644 +2,767 @@
...
+
+              <div class="card shadow">
+                <div class="card-header">
+                  <i class="fa fa-bug" aria-hidden="true"></i>
+                  <span data-i18n="optionsAboutReportBugs"></span>
+                </div>
+                <div class="card-body">
+                  <p>
+                    <a target="_blank" href="https://github.com/keepassxreboot/keepassxc-browser/issues/new" tabindex="14"><span data-i18n="optionsAboutCreateNewIssue"></span></a>
+                    <p><span data-i18n="optionsAboutDebugInfo"></span>:
```
Comment (xf-) [inline keepassxc-browser/options/options.html]: p in p is not allowed

**Pull request merged**